src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/CISShadowMonitor.java

```diff
@@ -81,54 +108,60 @@ public class CISShadowMonitor implements Consumer<NetworkStateProvider.Connectio
-        this(null, null, executorService, Coerce.toString(deviceConfiguration.getThingName()) + CIS_SHADOW_SUFFIX,
-                connectivityInformation);
+    public CISShadowMonitor(MqttClient mqttClient,
+                            NetworkStateProvider networkStateProvider,
+                            ExecutorService executorService,
+                            DeviceConfiguration deviceConfiguration,
+                            ConnectivityInformation connectivityInformation) {
+        this(
+                networkStateProvider,
+                null,
+                null,
+                executorService,
+                Coerce.toString(deviceConfiguration.getThingName()) + CIS_SHADOW_SUFFIX,
+                connectivityInformation,
+                mqttClient::getMqttOperationTimeoutMillis
+        );
         this.connection = new WrapperMqttClientConnection(mqttClient);
         this.iotShadowClient = new IotShadowClient(this.connection);
     }
 
-    CISShadowMonitor(MqttClientConnection connection, IotShadowClient iotShadowClient, ExecutorService executorService,
-                     String shadowName, ConnectivityInformation connectivityInformation) {
+    CISShadowMonitor(NetworkStateProvider networkStateProvider,
+                     MqttClientConnection connection,
+                     IotShadowClient iotShadowClient,
+                     ExecutorService executorService,
+                     String shadowName,
+                     ConnectivityInformation connectivityInformation,
+                     Supplier<Integer> mqttOperationTimeoutMillis) {
+        this.networkStateProvider = networkStateProvider;
         this.connection = connection;
         this.iotShadowClient = iotShadowClient;
         this.executorService = executorService;
         this.shadowName = shadowName;
         this.connectivityInformation = connectivityInformation;
+        this.mqttOperationTimeoutMillis = mqttOperationTimeoutMillis;
     }
 
     /**
      * Start shadow monitor.
      */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
     public void startMonitor() {
-        if (subscribeTaskFuture != null) {
-            subscribeTaskFuture.cancel(true);
-        }
-        subscribeTaskFuture = executorService.submit(() -> {
-            try {
-                subscribeToShadowTopics();
-                publishToGetCISShadowTopic();
-            } catch (InterruptedException e) {
-                LOGGER.atWarn().cause(e).log("Interrupted while subscribing to CIS shadow topics");
-                Thread.currentThread().interrupt();
-            }
-        });
+        fetchCISShadowAsync();
     }
 
     /**
      * Stop shadow monitor.
      */
     public void stopMonitor() {
-        if (subscribeTaskFuture != null) {
-            subscribeTaskFuture.cancel(true);
-        }
-        unsubscribeFromShadowTopics();
+        subscribed.set(false);
```

Comment:
why set it to false? You aren't ever unsubscribing, so you are still subscribed.

Reply:
revised the subscription stuff